s3.py

```diff
@@ -23,17 +23,48 @@ class S3Uploader(MirrorUploader):
 
     NAME = ExternalIntegration.S3
 
+    S3_HOSTNAME = "s3.amazonaws.com"
+    S3_BASE = "https://%s/" % S3_HOSTNAME
+
     BOOK_COVERS_BUCKET_KEY = u'book_covers_bucket'
     OA_CONTENT_BUCKET_KEY = u'open_access_content_bucket'
 
-    S3_HOSTNAME = "s3.amazonaws.com"
-    S3_BASE = "http://%s/" % S3_HOSTNAME
+    URL_TEMPLATE_KEY = u'bucket_name_transform'
+    URL_TEMPLATE_HTTP = u'http'
+    URL_TEMPLATE_HTTPS = u'https'
+    URL_TEMPLATE_DEFAULT = u'identity'
+
+    URL_TEMPLATES_BY_TEMPLATE = {
+        URL_TEMPLATE_HTTP: u'http://%(bucket)s/%(key)s',
+        URL_TEMPLATE_HTTPS: u'https://%(bucket)s/%(key)s',
+        URL_TEMPLATE_DEFAULT: S3_BASE + u'%(bucket)s/%(key)s',
+    }
 
     SETTINGS = [
         { "key": ExternalIntegration.USERNAME, "label": _("Access Key") },
         { "key": ExternalIntegration.PASSWORD, "label": _("Secret Key") },
-        { "key": BOOK_COVERS_BUCKET_KEY, "label": _("Book Covers Bucket"), "optional": True },
-        { "key": OA_CONTENT_BUCKET_KEY, "label": _("Open Access Content Bucket"), "optional": True },
+        { "key": BOOK_COVERS_BUCKET_KEY, "label": _("Book Covers Bucket"), "optional": True,
+          "description" : _("All book cover images encountered will be mirrored to this S3 bucket. Large images will be scaled down, and the scaled-down copies will also be uploaded to this bucket. <p>The bucket must already exist&mdash;it will not be created automatically.</p>")
```

Comment:
I thought we were going to have something to turn this on or off per collection - is that still true?

Reply:
There can still only one S3 integration, but you can choose whether or not to associate a given collection with it.